nginx-autoinstall.sh

```diff
@@ -1,8 +1,22 @@
 #!/bin/bash
 
-if [[ "$EUID" -ne 0 ]]; then
-	echo -e "Sorry, you need to run this as root"
-	exit 1
+# Detect absolute and full path as well as filename of this script
+cd "$(dirname $0)"
+CURRDIR=$(pwd)
+SCRIPT_FILENAME=$(basename $0)
+cd - > /dev/null
+sfp=$(readlink -f "${BASH_SOURCE[0]}" 2>/dev/null || greadlink -f "${BASH_SOURCE[0]}" 2>/dev/null)
+if [ -z "$sfp" ]; then sfp=${BASH_SOURCE[0]}; fi
+SCRIPT_DIR=$(dirname "${sfp}")
+
+# Make sure that the script runs with root permissions
+if [[ "$EUID" != 0 ]]; then
+  echo -e "Sorry, you need to run this as root. Please enter your root password...";
+  cd "$CURRDIR"
+  su -s "$(which bash)" -c "./$SCRIPT_FILENAME"
```

Comment:
I'm not a fan of this. The user should be smart enough to use sudo or to login as root. It's not worth adding 16 lines of hardly readable bash IMO

Reply:
Well, I'm not particularly a fan either, but it's convenient. 🤓

Reply:
Reverted the changes.